nvflare/app_opt/pt/quantization/quantizor.py

```diff
@@ -195,13 +202,34 @@ def process_dxo(self, dxo: DXO, shareable: Shareable, fl_ctx: FLContext) -> Unio
         """
 
         self.log_info(fl_ctx, "Running quantization...")
-        quantized_params, quant_state, source_datatype = self.quantization(params=dxo.data, fl_ctx=fl_ctx)
-        # Compose new DXO with quantized data
-        # Add quant_state to the new DXO meta
-        new_dxo = DXO(data_kind=dxo.data_kind, data=quantized_params, meta=dxo.meta)
-        new_dxo.set_meta_prop(key=MetaKey.PROCESSED_ALGORITHM, value=self.quantization_type)
-        new_dxo.set_meta_prop(key="quant_state", value=quant_state)
-        new_dxo.set_meta_prop(key="source_datatype", value=source_datatype)
-        self.log_info(fl_ctx, f"Quantized to {self.quantization_type}")
+
+        # for server job and already quantized message, skip quantization
+        # The reason is:
+        # server job in this case is 1-N communication with identical quantization operation
+        # the first communication to client will apply quantization and change the data on the server
+        # thus the subsequent communications to the rest of clients will no longer need to apply quantization
+        # This is not needed for client job, since the client job will be 1-1 and quantization applies to each client
+        # The behavior will also be different if each server-client filter is different, in which case
+        # a deep copy of the server data should be made by filter before applying the process
+        filter_flag = True
```

Comment:
Can we simplified to "if this filter is applied then don't apply again"?

Do we need to check if it is server side?

In the future this could be a general feature

Reply:
I think we need to provide more details for users to understand. Discussed this with Yan and Zhihong, and our conclusion is that this will be something to be handled by individual filters, not general pipeline. So we should also include this in our guidelines for filter design.

Reply:
got it, suggest rename to "should_apply_quantization"

Reply:
changed to quantized_flag to represent the state of the message and indicate a bool

Reply:
You already have "quantized_flag" what is that ?

Reply:
filter_flag only false when 1) server job 2) quantized_flag=True, so different var